developer_docs/provisioning_setup.asciidoc

```diff
@@ -0,0 +1,258 @@
+---
+:enabled: true
+:template_url: http://foreman.local.loc:8080
+
+# config/settings.d/tftp.yml
+---
+:enabled: true
+....
+
+[[foreman-configuration]]
+== Foreman configuration
+* _Infrastructure > Smart Proxies_ - Verify that all features are correctly loaded
+* _Infrastructure > Domains_ - Create `local.loc`, set DNS proxy to your local proxy
+* _Infrastructure > Subnets_ Create new subnet
+[source]
+....
+Name: default
+Network Address: 192.168.122.0
+Network Prefix: 24
+Network Mask: 255.255.255.0
+Gateway Address: 192.168.122.1
+Primary DNS server: 192.168.122.1
+IPAM: DHCP
+Start of IP range: 192.168.122.2
+End of IP range: 192.168.122.254
+MTU: 1500
+Boot mode: DHCP
+
+Proxies tab: Set all to your smart-proxy
+Domains tab: local.loc
+....
+* _Hosts > Operating Systems_ - Create new OS
+[source]
+....
+Name: CentOS_Stream
+Major Version: 9
+Family: Red Hat
+Architectures: x86_64
+Partition table: Kickstart default
+Installation media: CentOS Stream 9 mirror
+....
+* _Hosts > Provisioning templates_ - Assign templates to the OS
+  . PXELinux template: `Kickstart default PXELinux` (`name = "Kickstart default PXELinux"`)
+  . Provisioning template: `Kickstart default` (`name = "Kickstart default"`)
+  . Finish template: `Kickstart default finish` (`name = "Kickstart default finish"`)
+* _Hosts > Operating Systems_ - Set default templates and partition table for the OS
+* _Infrastructure > Compute Resources_ - Create compute resource for Libvirt
+[source]
+....
+Name: Libvirt
+Provider: Libvirt
+URL: qemu:///system
+# in case you want use ssh:
+# URL: qemu+ssh://your-account@192.168.122.1/system
+(Test connection)
+....
+
+[[provisioning-machine]]
+== Provisioning machine
+=== Machine in Libvirt
+Create new VM in Libvirt (Virtual Machine Manager)
+```
+How to install OS: Manuall install
+OS: CentOS Stream 9
+Memory: min 4096
+CPUs: min 2
+Customize configuration before install: yes
+```
+In configuration remember `MAC` address and set boot options to boot from network first.
+
+=== Create host in Foreman
+[source, bash]
+....
+# Host tab
+Deploy on: Bare metal
+
+# Operating system tab
+Architecture: x86_64
+OS: CentOS Stream 9
+Build mode: true
+Media: OS media
+Partition table: OS table
+
+Root password: changeme
+Provisioning templates: Check that all templates are loaded
+
+# Configure interface
+MAC Address: MAC from Libvirt
+Domain: local.loc
+IPv4 Subnet: default
+IPv4 Address: Check that IP has been assigned (Smart Proxy must be running)
+....
+
+After host is created in Foreman, restart VM in Virtual Network Manager and see the provisioning process.
+
+[[troubleshooting]]
+== Troubleshooting
+
+**Error when fetching kickstart from proxy**
+[source, bash]
+....
+Warning: anaconda: failed to fetch kickstart from http://your-smart-proxy/unattende ...
+....
+Solution: Check your firewall and verify that `virbr0` is in `trusted` zone
+
+**Permission denied while accessing `pxelinux.0`**
+
+Solution: Disable selinux.
```

Comment:
We should **never** recommend this. Instead, call `restorecon -R /var/lib/tftpboot` to properly label the files.